internal/app/machined/pkg/runtime/platform.go

```diff
@@ -28,6 +29,9 @@ type Platform interface {
 	// For metal platform that is either `talos.config=` URL or mounted ISO image.
 	Configuration(context.Context, state.State) ([]byte, error)
 
+	// Metadata fetches the platform metadata.
+	Metadata(context.Context, state.State) *runtime.PlatformMetadataSpec
```

Comment:
does this needs to be a pointer @smira ?

Reply:
refactoring -> runtime.PlatformMetadataSpec, error